src/monitoring_frame_deserialization.cpp

```diff
@@ -95,31 +119,80 @@ FieldHeader readFieldHeader(std::istringstream& is, const std::size_t& max_num_b
   {
     length--;
   }
-  return FieldHeader(id, length);
+  return MonitoringFrameAdditionalFieldHeader(id, length);
+}
+
+std::vector<MonitoringFrameDiagnosticMessage> deserializeDiagnosticMessages(std::istringstream& is)
+{
+  std::vector<MonitoringFrameDiagnosticMessage> diagnostic_messages;
+
+  std::array<uint8_t, DIAGNOSTIC_MESSAGE_UNUSED_OFFSET_IN_BYTES> reserved_diag_unused;
+  raw_processing::read(is, reserved_diag_unused);
+
+  for (auto& scanner_id : SCANNER_IDS)
```

Comment:
This breaks as soon as somebody adds an ID for Unknown etc...

Reply:
Thx. Fixed.